### PR TITLE
Languages & Frameworks: Add Gin guide (Golang)

### PR DESCRIPTION
**Pull request by unicodeveloper**
Completes Linear Issue [#LOG-2576](https://linear.app/railway/issue/LOG-2576).

CleanShot 2024-09-30 at 12 10 58@2x
CleanShot 2024-09-30 at 12 11 08@2x

### CODE REVIEW

File: src/docs/guides/gin.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+4. Run `railway up` to deploy.
+    - The CLI will now scan, compress and upload our fastapi app files to Railway's backend for deployment.
+
+## Use a Dockerfile
+
+1. Clone the forked `gin` repo and `cd` into the directory.
+    - You can skip this step if you already have an app directory or repo on your machine that you want to deploy.
+3. Create a `Dockerfile` in the `gin` or app's root directory.
+4. Add the content below to the `Dockerfile`:
+    ```bash
```
Comment (brody192) [inline src/docs/guides/gin.md:58]: better dockerfile -

```dockerfile
# Use the Go 1.23 alpine official image
# https://hub.docker.com/_/golang
FROM golang:1.23-alpine

# Create and change to the /app directory.
WORKDIR /app

# Copy go mod and sum files
COPY go.mod go.sum ./

# Install project dependencies
RUN go mod download

# Copy local code to the container image.
COPY . ./

# Build the app without debug symbols
RUN go build -ldflags="-w -s" -o app

# Run the app on container startup.
ENTRYPOINT ["./app"]
```

File: src/docs/guides/gin.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+width={2661} height={1019} quality={100} />
+
+## Deploy from the CLI
+
+1. <a href="/guides/cli#installing-the-cli" target="_blank">Install</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate with the CLI.</a>
+2. Clone the forked <a href="https://github.com/railwayapp-templates/gin" target="_blank">gin GitHub repo</a> and `cd` into the directory. 
+    - You can skip this step if you already have an app directory or repo on your machine that you want to deploy.
+3. Run `railway init` within the app directory to create a new project. 
+4. Run `railway up` to deploy.
+    - The CLI will now scan, compress and upload our fastapi app files to Railway's backend for deployment.
```
Comment (brody192) [inline src/docs/guides/gin.md:50]: gin*

**Review by brody192 — CHANGES_REQUESTED** [review]

**Comment by brody192** [comment]
I'll update the gin template repo, its quite old at this time.

File: src/docs/guides/gin.md (hunk 2 above)
Comment (unicodeveloper) [reply src/docs/guides/gin.md:50]: Good catch!

**Comment by unicodeveloper** [comment]
> I'll update the gin template repo, its quite old at this time.

I'll update to the latest gin release.

**Comment by brody192** [comment]
My bad, already had an update to the repo ready, feel free to modify it further to fir your needs.

File: src/docs/guides/gin.md (hunk 1 above)
Comment (unicodeveloper) [reply src/docs/guides/gin.md:58]: 1. Ah that makes sense. Thanks. 
2.  Exactly!

**Comment by unicodeveloper** [comment]
@brody192 Updated the fixes. Please take one more quick look so I can merge.

**Comment by brody192** [comment]
LGTM!

**Pull request merged**